ui/src/app/shared/requirements/form/requirements.form.html

```diff
@@ -1,52 +1,106 @@
+                    <sm-select class="search"
+                        *ngIf="availableRequirements.length > 0"
+                        (onChange)="selectType()"
+                        [(model)]="newRequirement.type"
+                        [options]="{'fullTextSearch': true}"
+                        placeholder="{{ 'requirement_type' | translate }}">
+                        <option *ngFor="let r of availableRequirements" value="{{r}}">{{r}}</option>
+                    </sm-select>
+                </div>
             </div>
+            <div class="five wide field">
+                <div *ngIf="newRequirement.type !== 'model'">
+                    <div class="field">
+                        <label>{{ 'requirement_value' | translate }}</label>
+                        <input
+                            name="value"
+                            auto-complete
+                            autocomplete="off"
+                            type="text"
+                            [source]="suggest"
+                            [min-chars]="3"
+                            [(ngModel)]="newRequirement.value"
+                            (keyup)="setName()"
+                            placeholder="{{'requirement_placeholder_value_' + newRequirement.type | translate}}"
+                            [no-match-found-text]="false">
+                    </div>
+                </div>
+                <div *ngIf="!loading && (newRequirement.type === 'model' || newRequirement.type === 'service')">
+                    <div class="field" *ngIf="newRequirement.type === 'model'">
+                        <label>{{'requirement_placeholder_value_' + newRequirement.type | translate}}</label>
+                        <sm-select
+                            class="search"
+                            [(model)]="newRequirement.value"
+                            [options]="{'fullTextSearch': true}"
+                            (onChange)="setName();"
+                            placeholder="{{'requirement_placeholder_value_' + newRequirement.type | translate}}">
+                            <option *ngFor="let m of workerModels" [value]="m.name">{{m.name}} ({{m.type}})</option>
+                        </sm-select>
+                    </div>
 
-
-        </div>
-        <div class="three wide center field">
-            <button class="ui blue icon button" (click)="addRequirement()"><i class="plus icon"></i></button>
-        </div>
+                    <div class="field" *ngIf="newRequirement.type === 'model'">
+                        <label>{{'requirement_manual_value' | translate}}</label>
+                        <input
+                            name="value"
+                            auto-complete
+                            autocomplete="off"
+                            type="text"
+                            [source]="suggest"
+                            [min-chars]="3"
+                            [(ngModel)]="newRequirement.value"
```

Comment:
Could be dangerous or have undefined behavior because if the value in the input doesn't exist in the list of select above. The value could be unsetted. Just a warning to know if you have tested this case.

Reply:
select removed, input only now.